### ISSUE TITLE
Native API allows you to create empty files with arbitrary checksum values and other metadata

### ISSUE THREAD

**Issue by pdurbin** [IQSS/dataverse#3413]
While testing the code that has lead to pull request #3412 I noticed that I'm able to persist to the database arbitrary information about files such as:
- label (filename)
- description
- checksum
- etc.

For what I expect is a full list please see file examples in https://github.com/IQSS/dataverse/blob/a509fcaa6c8b5d43c2b73ed1936b5586d25f20b7/src/test/resources/json/complete-dataset-with-files.json such as the one below.

`cat src/test/resources/json/complete-dataset-with-files.json | jq .datasetVersion.files[0]`

```
{
  "description": "",
  "label": "yn152_4_001.img",
  "directoryLabel": "data/subdir1/",
  "version": 1,
  "datasetVersionId": 2,
  "dataFile": {
    "id": 424,
    "filename": "yn152_4_001.img",
    "contentType": "application/octet-stream",
    "storageIdentifier": "data/subdir/yn152_4_001.img",
    "originalFormatLabel": "UNKNOWN",
    "checksum": {
      "type": "MD5",
      "value": "fc4ac06da6fd84785ed1b79cab9b714a73621ac7"
    },
    "description": ""
  }
}
```

Again, this was tested using the code in pull request #3412 (a509fca to be precise) but I have a very strong suspicion that we can expect this behavior in Dataverse 4.5 and/or 4.5.1 since @landreev indicated that this behavior was most likely introduced in the recent Harvesting effort (#813) in the part of the code that allows Harvesting to be done with what's called "native JSON" format at http://guides.dataverse.org/en/4.5.1/admin/metadataexport.html

In short, the files are there on the filesystem. End users don't have access to the file system so it doesn't make any sense for them to be able to create records in the database for files that don't exist on disk. If a user accidentally does this, it looks like they can easily delete these phony files without causing any weird exceptions, at least. We'll just hope they notice before they publish. 😄 

I mentioned this bug in passing in Slack and have been chatting with @landreev and @kcondon about it the most. They were in favor of me creating this issue.

**Comment by pdurbin** [comment]
This bug is somewhat alarming but not a high priority at the moment. Closing.
